android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java

```diff
@@ -266,7 +267,7 @@ private static void repeatNotification(Context context, NotificationDetails noti
                 repeatInterval = 60000 * 60;
                 break;
             case Daily:
-                repeatInterval = 60000 * 60 * 24;
+                repeatInterval = 60000 * 60 * 24 * dayInterval;
```

Comment:
A couple of refactoring/coding tips

1. `dayInterval` is only needed here so it's definition should be moved here to be more clear about its scope
2. having said that, it doesn't need to be declared and could be part of the expression for calculating the value of `repeatInterval` (see [here](https://refactoring.com/catalog/inlineVariable.html) where you may find a lot of useful refactoring techniques)

Reply:
Thanks for your feedback! I instantiated the `dayInterval` upfront to make it more clear and to keep the calculation of the `repeatInterval` easier to digest. But its scope is indeed clearer when moved to the Daily state. I've updated both the Android and iOS implementation.